internal/provider/data_source_test.go

```diff
@@ -38,6 +38,66 @@ func TestAccDataSource_inputs(t *testing.T) {
 	})
 }
 
+func TestAccTimeOffset_Upgrade_Basic(t *testing.T) {
```

Comment:
Please note that upgrade testing of data sources _generally_ does not have many benefits since Terraform core should always re-read data sources every execution. These tests can likely be removed unless we explicitly want to test Terraform core's functionality, not the provider upgrade (which I'd recommend doing in terraform-provider-corner rather than here).